### PR TITLE
Testing alternative notation for network definitions

### PR DESCRIPTION
**Pull request by dancrumb**
I went ahead and created this so that I could provide comments

### CODE REVIEW

File: core/Network.js [hunk 1]
```diff
@@ -97,11 +97,16 @@ Network.prototype.defProc = function (func, name) {
...
     throw new Error("No function passed to defProc");
   }
-  var proc = new Process(name || func.name, func);
-  trace('Created Process with name: ' + proc.name);
-
-  this._processes.push(proc);
-  return proc;
+  var s = name;
+  if (s == null)
+	  s = func.name;
```
Comment (dancrumb) [inline core/Network.js]: The problem with this is that func.name will not always be defined.

JavaScript supports anonymous functions:

`var foo = function () {}; // foo.name === undefined`

If you're going to use a process's name as the key in an associative array (which I think is a good idea), then we need an approach for default names.

File: core/runtimes/FiberRuntime/index.js [hunk 2]
```diff
@@ -139,20 +139,41 @@ FiberRuntime.prototype._hasDeadLock = function () {
...
-
+    
     if(selfstarting) {
-      queue.push(process);
+        queue.push(process);
+      }
+      */
+   // -----------------------------
+    var selfstarting = true;
+    for (var key in self._list) {
```
Comment (dancrumb) [inline core/runtimes/FiberRuntime/index.js]: I'm totally cool with this.

There's a library called lodash which I would recommend we use. It allows you to use `forEach` over any kind of collection, array or object, and avoids the boilerplate of having to call `hasOwnProperty`.

Comment (dancrumb) [inline core/runtimes/FiberRuntime/index.js]: But that can wait for now.

**Comment by jpaulm** [comment]
That's great! Thanks!  Not sure how the PR works with my new branch, but I guess I'll learn by doing... and by screwing up!  Seniors' moment: how do I incorporate lodash into this ever-growing construction?

**Comment by jpaulm** [comment]
Help!  I have added defproc checks to  alternative_network_defs and updated network fbptestxx and a couple of others.  `npm test` and Travis are now failing massively!  Where are the networks that are failing when I run `npm test`?

The biggie is that reader.js doesn't seem to be passing data back from the callback to the process (?) - it used to work!  It's been a while since I worked on this - I am wondering if JS has changed how callbacks are handled...  Could you do me a big favour and look at fbptest02.js and reader.js? (I have added a few console.logs to reader.js)

While I am working in a branch, what do I do about your other PRs - probably dumb question!

Lastly, we have to go out for the rest of the day - I'll have a little more time tomorrow morning, and then not much until Monday!  Sorry!

**Comment by dancrumb** [comment]
This is actually great!

The code that is failing is because the process name is not given in the tests... just as we wanted.

Those networks are in test/components/*.js

**Comment by dancrumb** [comment]
I fixed the tests that are failing due to a missing process name and pushed to this branch

**Comment by jpaulm** [comment]
That's great!  Thanks!

Have to leave for the day shortly - I'm really curious about reader.js -
hope you can figure it out!  That callback stuff was based on a suggestion
by Bruno Jouhier, and *was* working - does it still work in master?  Maybe
you could test it, and then we could go from there...?  I could also try to
find the original correspondence with Bruno...

Thanks,

Paul

**Comment by dancrumb** [comment]
I'm closing this as it's out of date. I'll open a new one

**Comment by dancrumb** [comment]
I opened #66 

As for the other PRs. If you're OK with them, just indicate with a comment and I can get them all merged (it may take a little massaging).

Otherwise, have a great weekend and we'll catch up on Monday.

**Pull request closed without merging**